src/main/cli/cli.c

```diff
@@ -1202,76 +1222,85 @@ static void printAux(dumpFlags_t dumpMask, const modeActivationCondition_t *mode
     }
 }
 
-static void cliAux(const char *cmdName, char *cmdline)
+static convertResult_t convertBoxPermanentIdToBoxId(long int permanentBoxId)
 {
-    int i, val = 0;
-    const char *ptr;
+    const box_t *box = findBoxByPermanentId(permanentBoxId);
+    if (!box) {
+        // cliPrintErrorLinef(cmdName, "BOX NOT MATCHED TO ANY BOX ID");
+        return (convertResult_t) { .isOk = false };
+    }
+    return (convertResult_t) { .isOk = true, .value = box->boxId };
+}
 
-    if (isEmpty(cmdline)) {
+static convertResult_t convertChannelValueToStep(long int channelValue)
+{
+    if (channelValue < CHANNEL_RANGE_MIN || channelValue > CHANNEL_RANGE_MAX) {
+        return (convertResult_t) { .isOk = false, .errMsg = "NOT BETWEEN 900 AND 2100" };
+    }
+    return (convertResult_t) { .isOk = true, .value = CHANNEL_VALUE_TO_STEP(channelValue) };
+}
+
+STATIC_UNIT_TESTED void cliAux(const char *cmdName, char *cmdline)
+{
+    parseArg_t ixArgs[] = {
```

Comment:
I also don't like it much. But it means we won't be able to assign parsed values to a field. The only way I'm able to see is using `offsetof`.

Reply:
`offsetof` and maybe dummy, stack allocated target may do.

It may even possible to infer destination type ( C11 `_Generic` ). But maybe that is too magical ...